armi/reactor/blueprints/blockBlueprint.py

```diff
@@ -241,6 +229,41 @@ def construct(
                 runLog.warning(str(e), single=True)
         return b
 
+    def _getBlockwiseMaterialModifierOptions(
+        self, children: Iterable[Composite]
+    ) -> Set[str]:
+        validMatModOptions = set()
+        for c in children:
+            perChildModifiers = self._getCompositeMaterialModifiers(c)
+            validMatModOptions.update(perChildModifiers)
+        return validMatModOptions
+
+    def _getCompositeMaterialModifiers(self, c: Composite) -> Set[str]:
+        perChildModifiers = set()
+        for material in self._getMaterialsInComposite(c):
+            for materialParentClass in material.__class__.__mro__:
+                # we must loop over parents as well, since applyInputParams
+                # could call to Parent.applyInputParams()
+                if issubclass(materialParentClass, Material):
+                    perChildModifiers.update(
+                        signature(
+                            materialParentClass.applyInputParams
+                        ).parameters.keys()
+                    )
+        # self is a parameter to class methods, so it gets picked up here
+        # but that's obviously not a real material modifier
+        perChildModifiers.discard("self")
+        return perChildModifiers
+
+    def _getMaterialsInComposite(self, child: Composite) -> Iterator[Material]:
```

Comment:
Ditto. Can this method get a docstring?

Reply:
There are some unit tests that use the composite group / component group feature. So there are things that currently live on a block that are not all _strict_ `Component`s.

This also looks toward particle fuel modeling (#1134) where there could be a non-`Component` living on the block